app/src/features/UserProfileSettings/index.tsx

```diff
@@ -0,0 +1,420 @@
+        const errors: ChangeUserNameErrors = {
+            firstName: undefined,
+            lastName: undefined,
+        };
+        if (changeUserNameRequest.firstName.trim().length === 0) {
+            errors.firstName = 'First name is required';
+        }
+        if (changeUserNameRequest.lastName.trim().length === 0) {
+            errors.lastName = 'Last name is required';
+        }
+        return errors;
+    }, [changeUserNameRequest]);
+
+    const hasNameValidationErrors = useMemo(() => {
+        return (
+            nameValidationErrors.firstName !== undefined ||
+            nameValidationErrors.lastName !== undefined
+        );
+    }, [nameValidationErrors]);
+
+    const hasNameChanged =
+        changeUserNameRequest.firstName !== firstName ||
+        changeUserNameRequest.lastName !== lastName;
+
+    const passwordValidationErrors = useMemo<ChangePasswordErrors>(() => {
+        const errors: ChangePasswordErrors = {
+            oldPassword: undefined,
+            newPassword: undefined,
+            repeatNewPassword: undefined,
+        };
+        if (
+            changePasswordRequest.oldPassword.trim().length === 0 &&
+            changePasswordRequest.newPassword.trim().length === 0 &&
+            changePasswordRequest.repeatNewPassword.trim().length === 0
+        ) {
+            return errors;
+        }
+        if (changePasswordRequest.oldPassword.trim().length === 0) {
+            errors.oldPassword = 'Old password is required';
+        }
+        if (changePasswordRequest.newPassword.trim().length === 0) {
+            errors.newPassword = 'New password is required';
+        }
+        if (changePasswordRequest.repeatNewPassword.trim().length === 0) {
+            errors.repeatNewPassword = 'Please repeat new password';
+        }
+        if (
+            changePasswordRequest.newPassword !==
+            changePasswordRequest.repeatNewPassword
+        ) {
+            errors.repeatNewPassword = 'Passwords do not match';
+        }
+        return errors;
+    }, [changePasswordRequest]);
+
+    const hasPasswordValidationErrors = useMemo(() => {
+        return (
+            passwordValidationErrors.oldPassword !== undefined ||
+            passwordValidationErrors.newPassword !== undefined ||
+            passwordValidationErrors.repeatNewPassword !== undefined
+        );
+    }, [passwordValidationErrors]);
+
+    const fullName = `${firstName} ${lastName}`;
+    const initials = `${firstName.length > 0 ? firstName[0] : ''}${
+        lastName.length > 0 ? lastName[0] : ''
+    }`;
+
+    const [changeNameRequestState, setChangeNameRequestState] =
+        useState<RequestState>({
+            status: 'idle',
+            errors: [],
+        });
+
+    const [changePasswordRequestState, setChangePasswordRequestState] =
+        useState<RequestState>({
+            status: 'idle',
+            errors: [],
+        });
+
+    const changeName = () => {
```

Comment:
Zar ovo ne izdvajamo kao neki redux ili nesto?

Reply:
iskreno, delovalo mi kao overkill koristiti redux za ovo. ova stanja (changeNameRequestErrors, itd) ce se koristiti samo na ovoj stranici i nigde vise, pa mi bzvz da idu u globalni state

Reply:
plus bi to povlacilo novi folder, novi reducer, jos thunkova itd :D